std/assembly/string.ts

```diff
@@ -413,6 +413,19 @@ export class String {
     return result;
   }
 
+  slice(beginIndex: i32, endIndex: i32 = i32.MAX_VALUE): String {
+    var length = this.length;
+    if (beginIndex < 0) beginIndex += length;
+    if (endIndex < 0) endIndex += length;
+    var from = min(max(beginIndex, 0), length);
+    var to = min(max(endIndex, 0), length);
```

Comment:
I think much better rewrite `beginIndex`/`endIndex` clamp like this:
```ts
begin = begin < 0 ? max<i32>(length + begin, 0) : min<i32>(begin, length);
end   = end   < 0 ? max<i32>(length + end,   0) : min<i32>(end,   length);
```

Reply:
Cool！